app/src/forms/form/service.js

```diff
@@ -416,6 +416,58 @@ const service = {
       .withGraphFetched('statusCode')
       .where('formId', formId);
   },
+
+  // -----------------------------------------------------------------------------
+  // API Key
+  // -----------------------------------------------------------------------------
+  // Modification actions are audited in form_api_key_audit via a trigger
+
+  // Get the current key for a form
+  readApiKey: async (formId) => {
+    return FormApiKey.query()
+      .modify('filterFormId', formId)
+      .first();
+  },
+
+  // Add an API key to the form, delete any existing key
+  createOrReplaceApiKey: async (formId, currentUser) => {
+    let trx;
+    try {
+      const currentKey = await service.readApiKey(formId);
+      trx = await transaction.start(FormApiKey.knex());
+
+      const newKey = {
+        formId: formId,
+        secret: uuidv4(),
+        createdBy: currentUser.username
+      };
```

Comment:
Can slipstrieam this object to line 451 as it's not used anywhere else. Also this constant doesn't need to be created so soon in order of operations, so can defer until line 451.